File: sdk/typescript/tests/client.test.ts
import { describe, it, expect } from "vitest";
import { EvixClient } from "../src/client.js";

describe("EvixClient", () => {
  it("should initialize with config", () => {
    const client = new EvixClient({
      chainId: 8453,
      contractAddress: "0x1234567890123456789012345678901234567890",
      publicClient: {} as any,
    });

    expect(client.chainId).toBe(8453);
    expect(client.contractAddress).toBe(
      "0x1234567890123456789012345678901234567890"
    );
  });

  it("should throw on swap without walletClient", async () => {
    const client = new EvixClient({
      chainId: 8453,
      contractAddress: "0x1234567890123456789012345678901234567890",
      publicClient: {} as any,
    });

    await expect(
      client.swap({
        tokenIn: "0x1111111111111111111111111111111111111111",
        tokenOut: "0x2222222222222222222222222222222222222222",
        amountIn: 1000n,
        deadline: 9999999999n,
        minAmountOut: 900n,
        recipient: "0x3333333333333333333333333333333333333333",
      })
    ).rejects.toThrow("WalletClient required for swap execution");
  });
});

File: sdk/rust/tests/client_test.rs
use alloy::primitives::{address, U256};
use evix_sdk::{EvixClient, SwapParams};

#[test]
fn test_client_creation() {
    let client = EvixClient::new(
        8453,
        address!("0x1234567890123456789012345678901234567890"),
    );
    assert_eq!(client.chain_id, 8453);
}

#[test]
fn test_encode_swap() {
    let client = EvixClient::new(
        8453,
        address!("0x1234567890123456789012345678901234567890"),
    );

    let params = SwapParams {
        token_in: address!("0x1111111111111111111111111111111111111111"),
        token_out: address!("0x2222222222222222222222222222222222222222"),
        amount_in: U256::from(1000u64),
        deadline: U256::from(9999999999u64),
        min_amount_out: U256::from(900u64),
        recipient: address!("0x3333333333333333333333333333333333333333"),
    };

    let calldata = client.encode_swap(&params);
    assert!(!calldata.is_empty());
}

#[tokio::test]
async fn test_swap_without_provider() {
    let client = EvixClient::new(
        8453,
        address!("0x1234567890123456789012345678901234567890"),
    );

    let params = SwapParams {
        token_in: address!("0x1111111111111111111111111111111111111111"),
        token_out: address!("0x2222222222222222222222222222222222222222"),
        amount_in: U256::from(1000u64),
        deadline: U256::from(9999999999u64),
        min_amount_out: U256::from(900u64),
        recipient: address!("0x3333333333333333333333333333333333333333"),
    };

    let result = client.swap(params).await;
    assert!(result.is_err());
}
